lib/components/col.js

```diff
@@ -0,0 +1,105 @@
+import { mergeData, uppercase } from "../utils";
+import { concat } from "../utils/array";
+import { keys, assign } from "../utils/object";
+
+function boolStrNum() {
+    return {
+        type: [Boolean, String, Number],
+        default: false
+    };
+}
+function strNum() {
+    return {
+        type: [String, Number],
+        default: null
+    };
+}
+
+function suffix(str, suffix) {
+    return suffix ? str + uppercase(suffix) : str;
+}
+
+function computeBkPt(prefix, bkpt, val) {
+    if (!val) {
+        return null;
+    }
+    if (val === true && bkpt) {
+        return `${prefix}-${bkpt}`;
+    }
+
+    return `${prefix}-${bkpt}-${val}`;
+}
+
+const BKPTS = ["sm", "md", "lg", "xl"];
+// This generates plain [offset, push, pull] props without bkpt suffixes.
+const BKPTS_PADDED = concat(null, BKPTS);
+
+const breakpointProps = BKPTS.reduce((memo, bkpt) => ((memo[bkpt] = boolStrNum()), memo), {});
+const offsetProps = BKPTS_PADDED.reduce((memo, bkpt) => ((memo[suffix("offset", bkpt)] = strNum()), memo), {});
+const pushProps = BKPTS_PADDED.reduce((memo, bkpt) => ((memo[suffix("push", bkpt)] = strNum()), memo), {});
+const pullProps = BKPTS_PADDED.reduce((memo, bkpt) => ((memo[suffix("pull", bkpt)] = strNum()), memo), {});
+
+const propKeys = [
+    ["col", keys(breakpointProps)],
+    ["offset", keys(offsetProps)],
+    ["push", keys(pushProps)],
+    ["pull", keys(pullProps)]
+];
+
+export const props = assign(
+    {
+        tag: {
+            type: String,
+            default: "div"
+        },
+        col: {
+            type: Boolean,
+            default: false
+        },
+        size: {
+            type: [String, Number],
+            default: null
+        }
```

Comment:
I m thinking `size` prop should be called `cols`, so as to not confuse/conflict with the `size` prop used on many other components (i.e. button, button-group, etc) which refers to values of `sm`, `lg` and `md` for button sizes.

b-form-fieldset now uses the `cols` prop instead of `size` to prevent user confusion (a few issues were generated for-form-fieldset because of confusion as to the value)